examples/rust/battleship/core/src/lib.rs

```diff
@@ -33,19 +33,19 @@ pub const SHIP_SPANS: [usize; NUM_SHIPS] = [
 
 pub const SUNK_MASKS: [u8; NUM_SHIPS] = [0x1f, 0x0f, 0x07, 0x07, 0x03];
 
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, Hash)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, Hash, Copy)]
```

Comment:
I'd prefer copies to be explicit for structs generally. There's about 3 options I can think of when rust complains about something relating to borrow checking:

1. Implement `Copy` (implicit clone).
2. Explicitly call `.clone()`.
3. Pass by ref with a borrow than by value with a clone/copy.

My assumption is that you wanted to have `Copy` because the compiler complained, but I'd suggest trying to use `.clone()` where it was complaining.